### DISCUSSION TITLE
Importing re-exported modules from an index.ts file also imports unused modules

### DISCUSSION THREAD

**Discussion by resthedev** [vercel/next.js#49121] [Help]
### Summary

I have been experiencing issues when importing re-exported modules from `index.ts` files.

For example, my `index.ts` file inside `@/features/my-feature` looks like this:
```tsx
export { ComponentA } from "./components/ComponentA";
export { ComponentB } from "./components/ComponentB";
export { ComponentC } from "./components/ComponentC";
```

Then in another file, if I do:
```
import { ComponentA } from "@/features/my-feature;
```

...it also is importing `ComponentB` and `ComponentC` alongside `ComponentA` (even when those are completely unused).

Is this expected behavior? 

I was expecting for some form of tree shaking to occur and for the unused imports to not also be inadvertently imported as well.

**Comment by Poolshark** [comment]
I have encountered the same issue! Is there any progress on this?

**Comment by lykianovsky** [comment]
https://github.com/vercel/next.js/issues/69532
